src/gpuarray_buffer_cuda.c

```diff
@@ -910,6 +973,36 @@ static gpukernel *cuda_newkernel(void *c, unsigned int count,
           cuda_exit(ctx);
           return NULL;
         }
+        if (compile_cache == NULL)
```

Comment:
I would move this to before line 933, otherwise, sometimes the cache won't be defined.